### PR TITLE
Bump AWS SDK to v1.17.4

### PR DESCRIPTION
**Pull request by ewbankkit**
[AWS SDK v1.17.4](https://github.com/aws/aws-sdk-go/releases/tag/v1.17.4) required for:
* https://github.com/terraform-providers/terraform-provider-aws/issues/7673
* https://github.com/terraform-providers/terraform-provider-aws/issues/7672
* https://github.com/terraform-providers/terraform-provider-aws/issues/7671
* https://github.com/terraform-providers/terraform-provider-aws/issues/7668
* https://github.com/terraform-providers/terraform-provider-aws/issues/7667
* https://github.com/terraform-providers/terraform-provider-aws/issues/7662
* https://github.com/terraform-providers/terraform-provider-aws/issues/7650

```console
$ go get github.com/aws/aws-sdk-go@v1.17.4
$ go mod tidy
$ go mod vendor
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
_WIP_ because

```console
$ make
==> Checking that code complies with gofmt requirements...
go install
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_glue_crawler.go:369:47: cannot convert *crawlerInput (type glue.CreateCrawlerInput) to type glue.UpdateCrawlerInput
GNUmakefile:10: recipe for target 'build' failed
make: *** [build] Error 2
```

which is the same as https://github.com/terraform-providers/terraform-provider-aws/issues/7661.

**Comment by ewbankkit** [comment]
Cherry picked @YakDriver 's commit from https://github.com/terraform-providers/terraform-provider-aws/pull/7660.
Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGlueCrawler_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSGlueCrawler_ -timeout 120m
=== RUN   TestAccAWSGlueCrawler_DynamodbTarget
=== PAUSE TestAccAWSGlueCrawler_DynamodbTarget
=== RUN   TestAccAWSGlueCrawler_JdbcTarget
=== PAUSE TestAccAWSGlueCrawler_JdbcTarget
=== RUN   TestAccAWSGlueCrawler_JdbcTarget_Exclusions
=== PAUSE TestAccAWSGlueCrawler_JdbcTarget_Exclusions
=== RUN   TestAccAWSGlueCrawler_JdbcTarget_Multiple
=== PAUSE TestAccAWSGlueCrawler_JdbcTarget_Multiple
=== RUN   TestAccAWSGlueCrawler_S3Target
=== PAUSE TestAccAWSGlueCrawler_S3Target
=== RUN   TestAccAWSGlueCrawler_S3Target_Exclusions
=== PAUSE TestAccAWSGlueCrawler_S3Target_Exclusions
=== RUN   TestAccAWSGlueCrawler_S3Target_Multiple
=== PAUSE TestAccAWSGlueCrawler_S3Target_Multiple
=== RUN   TestAccAWSGlueCrawler_recreates
=== PAUSE TestAccAWSGlueCrawler_recreates
=== RUN   TestAccAWSGlueCrawler_Classifiers
=== PAUSE TestAccAWSGlueCrawler_Classifiers
=== RUN   TestAccAWSGlueCrawler_Configuration
=== PAUSE TestAccAWSGlueCrawler_Configuration
=== RUN   TestAccAWSGlueCrawler_Description
=== PAUSE TestAccAWSGlueCrawler_Description
=== RUN   TestAccAWSGlueCrawler_Role_ARN_NoPath
=== PAUSE TestAccAWSGlueCrawler_Role_ARN_NoPath
=== RUN   TestAccAWSGlueCrawler_Role_ARN_Path
=== PAUSE TestAccAWSGlueCrawler_Role_ARN_Path
=== RUN   TestAccAWSGlueCrawler_Role_Name_Path
=== PAUSE TestAccAWSGlueCrawler_Role_Name_Path
=== RUN   TestAccAWSGlueCrawler_Schedule
=== PAUSE TestAccAWSGlueCrawler_Schedule
=== RUN   TestAccAWSGlueCrawler_SchemaChangePolicy
=== PAUSE TestAccAWSGlueCrawler_SchemaChangePolicy
=== RUN   TestAccAWSGlueCrawler_TablePrefix
=== PAUSE TestAccAWSGlueCrawler_TablePrefix
=== RUN   TestAccAWSGlueCrawler_SecurityConfiguration
=== PAUSE TestAccAWSGlueCrawler_SecurityConfiguration
=== CONT  TestAccAWSGlueCrawler_DynamodbTarget
=== CONT  TestAccAWSGlueCrawler_SecurityConfiguration
=== CONT  TestAccAWSGlueCrawler_TablePrefix
=== CONT  TestAccAWSGlueCrawler_SchemaChangePolicy
=== CONT  TestAccAWSGlueCrawler_Schedule
=== CONT  TestAccAWSGlueCrawler_Role_Name_Path
=== CONT  TestAccAWSGlueCrawler_Role_ARN_Path
=== CONT  TestAccAWSGlueCrawler_Role_ARN_NoPath
=== CONT  TestAccAWSGlueCrawler_Description
=== CONT  TestAccAWSGlueCrawler_Configuration
=== CONT  TestAccAWSGlueCrawler_Classifiers
=== CONT  TestAccAWSGlueCrawler_recreates
=== CONT  TestAccAWSGlueCrawler_S3Target_Multiple
=== CONT  TestAccAWSGlueCrawler_S3Target_Exclusions
=== CONT  TestAccAWSGlueCrawler_S3Target
=== CONT  TestAccAWSGlueCrawler_JdbcTarget_Multiple
=== CONT  TestAccAWSGlueCrawler_JdbcTarget_Exclusions
=== CONT  TestAccAWSGlueCrawler_JdbcTarget
--- PASS: TestAccAWSGlueCrawler_Role_ARN_NoPath (45.08s)
--- PASS: TestAccAWSGlueCrawler_SchemaChangePolicy (46.30s)
--- PASS: TestAccAWSGlueCrawler_recreates (53.56s)
--- PASS: TestAccAWSGlueCrawler_Role_Name_Path (56.39s)
--- PASS: TestAccAWSGlueCrawler_Classifiers (62.40s)
--- PASS: TestAccAWSGlueCrawler_Role_ARN_Path (63.33s)
--- PASS: TestAccAWSGlueCrawler_JdbcTarget (65.38s)
--- PASS: TestAccAWSGlueCrawler_DynamodbTarget (67.35s)
--- PASS: TestAccAWSGlueCrawler_Description (68.29s)
--- PASS: TestAccAWSGlueCrawler_JdbcTarget_Multiple (74.79s)
--- PASS: TestAccAWSGlueCrawler_TablePrefix (75.32s)
--- PASS: TestAccAWSGlueCrawler_Schedule (76.31s)
--- PASS: TestAccAWSGlueCrawler_S3Target_Multiple (87.11s)
--- PASS: TestAccAWSGlueCrawler_JdbcTarget_Exclusions (96.67s)
--- PASS: TestAccAWSGlueCrawler_Configuration (97.83s)
--- PASS: TestAccAWSGlueCrawler_S3Target_Exclusions (98.83s)
--- PASS: TestAccAWSGlueCrawler_S3Target (101.73s)
--- PASS: TestAccAWSGlueCrawler_SecurityConfiguration (140.37s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	140.412s
```

File: go.sum [hunk 1]
```diff
@@ -10,6 +10,8 @@ github.com/armon/go-radix v1.0.0 h1:F4z6KzEeeQIMeLFa97iZU6vupzoecKdU5TX24SNppXI=
 github.com/armon/go-radix v1.0.0/go.mod h1:ufUuZ+zHj4x4TnLV4JWEpy2hxWSpsRywHrMgIH9cCH8=
 github.com/aws/aws-sdk-go v1.17.0 h1:+pbWEdKxH1qlLb07as1+auEVvx+IxkaDzQLwMzbK1tI=
```
Comment (bflad) [inline go.sum]: Hmm. When I ran `go mod tidy` the v1.17.0 were removed. We'll likely be setting up renovatebot to handle this going forward so hopefully the Go Modules quirks will be reduced. 🤞

**Comment by YakDriver** [comment]
Underbar! 🎉

**Pull request merged**